pkg/controller/kubelet-config/kubelet_config_bootstrap_test.go

```diff
@@ -0,0 +1,196 @@
+	require.NoError(t, err)
+	regfile := ignCfg.Storage.Files[0]
+	conf, err := dataurl.DecodeString(*regfile.Contents.Source)
+	require.NoError(t, err)
+	require.Contains(t, string(conf.Data), `"maxPods": 100`)
+}
+
+func TestGenerateDefaultManagedKeyKubelet(t *testing.T) {
+	workerPool := helpers.NewMachineConfigPool("worker", nil, helpers.WorkerSelector, "v0")
+	masterPool := helpers.NewMachineConfigPool("master", nil, helpers.WorkerSelector, "v0")
+	kcRaw, err := EncodeKubeletConfig(&kubeletconfigv1beta1.KubeletConfiguration{MaxPods: 100}, kubeletconfigv1beta1.SchemeGroupVersion)
+	if err != nil {
+		panic(err)
+	}
+
+	// valid case, only 1 kubeletconfig per pool
+	managedKeyExist := make(map[string]bool)
+	for _, tc := range []struct {
+		kubeletconfig      *mcfgv1.KubeletConfig
+		pool               *mcfgv1.MachineConfigPool
+		expectedManagedKey string
+	}{
+		{
+			&mcfgv1.KubeletConfig{
+				ObjectMeta: metav1.ObjectMeta{Name: "kcfg-default"},
+				Spec: mcfgv1.KubeletConfigSpec{
+					KubeletConfig: &runtime.RawExtension{
+						Raw: kcRaw,
+					},
+				},
+			},
+			workerPool,
+			"99-worker-generated-kubelet",
+		},
+		{
+			&mcfgv1.KubeletConfig{
+				ObjectMeta: metav1.ObjectMeta{Name: "kcfg-default"},
+				Spec: mcfgv1.KubeletConfigSpec{
+					KubeletConfig: &runtime.RawExtension{
+						Raw: kcRaw,
+					},
+				},
+			},
+			masterPool,
+			"99-master-generated-kubelet", // kubeletconfig apply to master pool, expected managedKey for master pool
+		},
+	} {
+		res, err := generateBootstrapManagedKeyKubelet(tc.pool, managedKeyExist)
+		require.NoError(t, err)
+		require.Equal(t, tc.expectedManagedKey, res)
+	}
+
+	// error case, 2 kubeletconfig applied for master pool
+	managedKeyExist = make(map[string]bool)
+	for _, tc := range []struct {
+		kubeletconfig      *mcfgv1.KubeletConfig
+		pool               *mcfgv1.MachineConfigPool
+		expectedManagedKey string
+		expectedErr        error
+	}{
+		{
+			&mcfgv1.KubeletConfig{
+				ObjectMeta: metav1.ObjectMeta{Name: "kcfg-default"},
+				Spec: mcfgv1.KubeletConfigSpec{
+					KubeletConfig: &runtime.RawExtension{
+						Raw: kcRaw,
+					},
+				},
+			},
+			workerPool,
+			"99-worker-generated-kubelet",
+			nil,
+		},
+		{
+			&mcfgv1.KubeletConfig{
+				ObjectMeta: metav1.ObjectMeta{Name: "kcfg-default"},
+				Spec: mcfgv1.KubeletConfigSpec{
+					KubeletConfig: &runtime.RawExtension{
+						Raw: kcRaw,
+					},
+				},
+			},
+			masterPool,
+			"99-master-generated-kubelet", // kubeletconfig apply to master pool, expected managedKey for master pool
+			nil,
+		},
+		{
+			&mcfgv1.KubeletConfig{
+				ObjectMeta: metav1.ObjectMeta{Name: "kcfg-1"},
+				Spec: mcfgv1.KubeletConfigSpec{
+					KubeletConfig: &runtime.RawExtension{
+						Raw: kcRaw,
+					},
+				},
+			},
+			masterPool,
+			"",
+			fmt.Errorf("error finding multiple kubeletconfigs for the same pool master. Please apply one kubeletconfig manifest for each pool"),
```

Comment:
If you update the error message above, don't forget about this one as well. That said, it might trip someone up later to hard code this expected error message and expect the error message to be changed in two locations.